harness/determined/common/schemas/expconf/_gen.py

```diff
@@ -2637,13 +2637,30 @@
     "$id": "http://determined.ai/schemas/expconf/v0/searcher-custom.json",
     "title": "CustomConfig",
     "type": "object",
-    "additionalProperties": false,
+    "additionalProperties": true,
```

Comment:
this is probably the most potentially controversial part of this change. I have to do it because "custom" does not support `source_trial_id` and `source_checkpoint_uuid`. I am happy to revisit this later.